articles/app/Http/Controllers/PostController.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Post;
+use App\Models\Category;
+use App\Models\Tag;
+use File;
+
+class PostController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $post = Post::OrderBy('id','DESC')->get();
+        return view('post.index',compact('post'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $category = Category::get();
+        $tag =Tag::get();
+        return view('post.create',compact('category','tag'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $request->validate([
+
+            'title' => 'required',
+            'description' => 'required',
+        ]);
+
+
+        $store_image = null;
+        if (request()->hasFile('image')) 
+        {
+            $file = request()->File('image');
+            $store_image = md5($file->getClientOriginalName()) . time() . "." . $file->getClientOriginalExtension();
+            $file->move('./images/', $store_image);
+        } 
+
+        $data = array(
+            'title' => $request->title,
+            'description' => $request->description,
+            'category_id' => $request->category_id,
+            'tag_id' => implode("," , $request->tag_id),
+            'image'       =>   $store_image,
+        );
+
+        Post::create($data);
+        return redirect('post');
```

Comment:
@FareenaEhtesham I would advice to use Route Names here: https://laravel.com/docs/8.x/routing#named-routes